src/types/routeWithParams.ts

```diff
@@ -0,0 +1,16 @@
+import { RegisteredRoutes } from '@/types/register'
+import { RouteMethods } from '@/types/routeMethods'
+import { ExtractRoutePathParameters, RoutePaths } from '@/types/routePaths'
+import { Routes } from '@/types/routes'
+
+export type RouteWithParams<
+  TRoutes extends Routes,
+  TRoutePath extends string
+> = {
+  route: TRoutePath & RoutePaths<TRoutes>,
+  params: ExtractRoutePathParameters<RouteMethods<TRoutes>, TRoutePath>,
+}
+
+export type RegisteredRouteWithParams<T extends string> = RouteWithParams<RegisteredRoutes, T>
```

Comment:
Quick note on this type. I didn't have this type originally and just used `RouteWithParams<RegisteredRoutes, T>` directly in RouterLink. However when inspecting the types output in the build vite was infering the `[]` from `RegisteredRoutes` and inserting that directly. Which means the registered routes would never be used. Probably related to vue-tsc's custom implementation of defineProp typs. Not using `RegisteredRoutes` directly helped keep it in tact.